java/algorithm/src/main/java/org/apache/arrow/algorithm/dictionary/HashTableBasedDictionaryBuilder.java

```diff
@@ -19,9 +19,9 @@
 
 import java.util.HashMap;
 
-import org.apache.arrow.memory.util.ArrowBufPointer;
 import org.apache.arrow.memory.util.hash.ArrowBufHasher;
 import org.apache.arrow.memory.util.hash.SimpleHasher;
+import org.apache.arrow.util.ArrowBufPointer;
```

Comment:
let's put in algorithm or similar type of package rather than a generalized util package.